src/StudentsController.ts

```diff
@@ -165,7 +169,36 @@ export class StudentsController {
         await this.#studentsCollection.update(oldDoc, student.toJSON());
     }
 
+    public async pseudonymizeStudent(student: Student): Promise<Student> {
+        // Only keep id and status of student (also works for future student properties)
+        const oldDoc = await this.#studentsCollection.read(student.id.toString());
+        const pseudonymizedDoc = {
+            id: oldDoc.id,
+            status: "deleted"
+        };
```

Comment:
It feels bad that all the properties are nullable now, even though they are filled in most cases.
Maybe a separate collection for deleted students may be better? 🤔